packages/components/src/box-control/test/index.tsx

```diff
@@ -325,19 +286,21 @@ describe( 'BoxControl', () => {
 
 			render( <BoxControl onChange={ setState } /> );
 
+			// Typing the first letter of a unit blurs the input.
 			await user.type(
 				screen.getByRole( 'textbox', {
 					name: 'Box Control',
 				} ),
-				'7.5rem'
+				'7r'
 			);
-			await user.keyboard( '{Enter}' );
 
-			expect( setState ).toHaveBeenCalledWith( {
-				top: '7.5rem',
-				right: '7.5rem',
-				bottom: '7.5rem',
-				left: '7.5rem',
+			// Due to the test environment not being a browser the units here
```

Comment:
Exactly what part of the implementation doesn't work in the test environment? It would be great if we had unit tests for this new behavior

Reply:
The actual unit switching is entirely up to the browser. The component merely moves the focus and what seems to be de-facto standard behavior of browsers changes the select according to key presses. We could cover it with e2e tests.